### ISSUE TITLE
[sfntdiff/sfntedit/spot] Fatal failures on Linux

### ISSUE THREAD

**Issue by miguelsousa** [adobe-type-tools/afdko#570]
Every test of `sfntdiff`, `sfntedit` and `spot` is failing on Linux.
The error message is the same on all of them:
```
[FATAL]: bad input object size [8]
```

**Comment by miguelsousa** [comment]
Examples of failing commands:
```
$ sfntdiff tests/sfntdiff_data/input/regular.otf tests/sfntdiff_data/input/bold.otf
```
```
$ sfntedit -x GDEF=foo tests/sfntedit_data/input/light.otf
$ sfntedit -a GDEF=tests/sfntedit_data/input/GDEF_italic.tb tests/sfntedit_data/input/italic.otf
```
```
$ spot -T tests/spot_data/input/black.otf
$ spot -F tests/spot_data/input/black.otf
```

**Comment by miguelsousa** [comment]
Fixed by #579

**Comment by miguelsousa** [comment]
`test_cid_keyed_cff_bug470` (from makeotf_test.py) is still failing on Linux.
I narrowed the source of the failure to a call to **sfntedit** in `convertfonttocid.mergeFontToCFF`.
This is the error message:
```
sfntedit [FATAL]: file rename error <Invalid cross-device link> [/tmp/tmpQqcwJp]
```
https://travis-ci.org/adobe-type-tools/afdko/jobs/420674923

I wasn't able to reproduce this in my Ubuntu 14 VM.
See `linux-tests` branch.

**Comment by khaledhosny** [comment]
Looks like https://stackoverflow.com/questions/42392600/oserror-errno-18-invalid-cross-device-link

**Comment by cjchapman** [comment]
@miguelsousa Since the `rename` error is different from the `bad input object size` error that this issue was about, I suggest closing this issue and starting a new one.

@khaledhosny Thanks for the info.

**Comment by miguelsousa** [comment]
@cjchapman it's all part of the same goal, so let's just use this one. This was all the information is kept in one place

**Comment by cjchapman** [comment]
We use the Python `tempfile.mkstemp()` in both the tests and the production code, so it's not as straightforward to change it in Python as I thought. I'd prefer to keep using `rename()` in C, but it seems like we may need to implement a `move()` that works across file systems.

**Comment by cjchapman** [comment]
OK, it turns out that changing the calls to `tempfile.mkstemp()` in `convertfonttocid_test.py` and `makeotf_test.py` to use a temporary directory on the same file system as the test data was sufficient.

**Comment by miguelsousa** [comment]
`file rename error` fixed by 76f5b65abd3f683135b4dc8176378c36a3e1d2cf
